linker/jvm/src/main/scala/org/scalajs/linker/PathOutputFile.scala

```diff
@@ -35,8 +37,28 @@ object PathOutputFile {
 
   private final class AtomicPathOutputFileImpl(path: Path) extends OutputFileImpl {
     def writeFull(buf: ByteBuffer)(implicit ec: ExecutionContext): Future[Unit] = {
-      Future(blocking(Files.createTempFile(
-          path.getParent(), ".tmp-" + path.getFileName(), ".tmp"))).flatMap { tmpFile =>
+      Future {
+        val tmpFile = blocking(Files.createTempFile(
+            path.getParent(), ".tmp-" + path.getFileName(), ".tmp"))
+
+        try {
+          /* Set file permissions for temporary file as in Linux it is created
+           * with permissions 0600 which deviates from the standard 0644 used
+           * in *.class and *.jar files.
+           * 
+           * Uses setPosixFilePermissions instead of passing the permissions to
+           * createTempFile to cover systems (if any exist) that uses POSIX
+           * permissions, but can't set them atomically while creating a
+           * temporary file.
+           */
+          val permissions = PosixFilePermissions.fromString("rw-r--r--")
+          blocking(Files.setPosixFilePermissions(tmpFile, permissions))
```

Comment:
I don't think there's any point in having two independent `blocking()` calls here. Consider wrapping one `blocking { ... }` around both the `createTempFile` and the `setPosixFilePermissions`.

Reply:
Okay, I will.